polling_stations/apps/file_uploads/views.py

```diff
@@ -0,0 +1,137 @@
+import json
+from datetime import datetime
+
+from django.conf import settings
+from django.contrib.admin.views.decorators import staff_member_required
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.core.exceptions import ValidationError as DjangoValidationError
+from django.db.models import Q
+from django.http import JsonResponse
+from django.views.decorators.csrf import csrf_protect, ensure_csrf_cookie
+from django.utils.decorators import method_decorator
+from django.views.generic import TemplateView
+
+import boto3
+from boto.pyami.config import Config
+from botocore.exceptions import ClientError
+from marshmallow import Schema, fields, validate
+from marshmallow import ValidationError as MarshmallowValidationError
+
+from councils.models import Council
+
+
+MAX_FILE_SIZE = 150000000  # 150Mb
+S3_BUCKET_NAME = "pollingstations-data-dev"
+
+
+class FileSchema(Schema):
+    name = fields.String(
+        required=True,
+        validate=validate.Regexp(
+            r"(?i)^.+(\.csv|\.tsv)$", error="Unexpected file type"
+        ),
+    )
+    size = fields.Integer(
+        required=True,
+        validate=validate.Range(min=1, max=MAX_FILE_SIZE, error="File too big"),
+    )
+    type = fields.String(
+        required=True,
+        validate=validate.Regexp(r"^text\/.+$", error="Unexpected file type"),
+    )
+
+
+class UploadRequestSchema(Schema):
+    council_id = fields.String(required=True)
+    files = fields.List(
+        fields.Nested(FileSchema()),
+        required=True,
+        validate=validate.Length(min=1, max=2),
+    )
+
+
+def get_s3_client():
+    config = Config()
+    access_key = config.get_value(settings.BOTO_SECTION, "aws_access_key_id")
+    secret_key = config.get_value(settings.BOTO_SECTION, "aws_secret_access_key")
+    return boto3.client(
+        "s3", aws_access_key_id=access_key, aws_secret_access_key=secret_key
+    )
+
+
+class FileUploadView(LoginRequiredMixin, TemplateView):
+    template_name = "file_uploads/index.html"
+
+    @method_decorator(staff_member_required)
+    @method_decorator(ensure_csrf_cookie)
+    def get(self, request, *args, **kwargs):
+        return super().get(self, request, *args, **kwargs)
+
+    def get_context_data(self, **context):
+        context["councils"] = (
+            Council.objects.all()
+            .filter(~Q(council_id__startswith="N09"))
+            .defer("area")
+            .order_by("name")
+        )
+        return context
+
+    def validate_body(self, body):
+        """
+        Do some basic picture checks on request body and files.
+        Obviously these values can be spoofed, so checks on
+        file extension and MIME type etc are just to "fail fast".
+        We will check the files _properly_ in the lambda hook.
+        """
+        try:
+            UploadRequestSchema().load(body)
+        except MarshmallowValidationError:
+            raise DjangoValidationError("Request body did not match schema")
```

Comment:
TODO: it would be really nice here to report a more meaningful error but its also a pain to parse `e.messages` in a general way. I'm going to leave it for now. We'll swallow it on the front-end anyway.